File: windows/Scanner/Rules/AESCFG001.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESCFG001 : IRule
{
    public string Id => "AES-CFG-001";
    public Severity Severity => Severity.High;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts) =>
        facts.Settings
            .Where(setting => setting.AppId == "cursor" && setting.Key == "cursor.privacyMode")
            .Where(setting => setting.BoolValue == false)
            .Select(setting => this.Finding(setting.AppId, affectedPath: setting.ConfigPath))
            .ToArray();
}

File: macos/Scanner/Sources/Scanner/Rules/AESCFG001.swift
public struct AESCFG001: Rule {
    public let id = "AES-CFG-001"
    public let severity: Severity = .high

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        facts.settings.compactMap { setting in
            guard setting.appId == "cursor", setting.key == "cursor.privacyMode" else {
                return nil
            }
            guard setting.boolValue == false else {
                return nil
            }
            return finding(app: setting.appId, affectedPath: setting.configPath)
        }
    }
}
